nextcord/embeds.py

```diff
@@ -427,17 +455,79 @@ def set_image(self: E, url: MaybeEmpty[Any]) -> E:
         -----------
         url: :class:`str`
             The source URL for the image. Only HTTP(S) is supported.
+        file: Optional[File]
+            A file to use for the image.
+
+            .. versionadded:: 2.0
         """
+        from . import InvalidArgument
 
-        if url is EmptyEmbed:
+        if url is EmptyEmbed and not file:
             try:
                 del self._image
             except AttributeError:
                 pass
+            try:
+                del self._local_files["image"]
+            except:
+                pass
+
+        elif url and file:
+            raise InvalidArgument("Can't pass both url and file.")
+
+        elif file:
+            self._local_files["image"] = file
+            self._image = {"url": f"attachment://{file.filename}"}
+
         else:
-            self._image = {
-                "url": str(url),
-            }
+            self._image = {"url": str(url)}
+
+        return self
+
+    def set_author(
+        self: E,
+        *,
+        name: Any,
+        file: Optional[File] = None,
+        url: MaybeEmpty[Any] = EmptyEmbed,
+        icon_url: MaybeEmpty[Any] = EmptyEmbed,
+    ) -> E:
+        """Sets the author for the embed content.
+
+        This function returns the class instance to allow for fluent-style
+        chaining.
+
+        Parameters
+        -----------
+        name: :class:`str`
+            The name of the author.
+        url: :class:`str`
+            The URL for the author.
+        icon_url: :class:`str`
+            The URL of the author icon. Only HTTP(S) is supported.
+        file: Optional[File]
+            A file to use for the image.
+
+            .. versionadded:: 2.0
+        """
+        from . import InvalidArgument
+
+        self._author = {
+            "name": str(name),
```

Comment:
Can probably fit this on 1 line
```suggestion
            "name": str(name)
```